File: 2526_find_consecutive_integers_from_a_data_stream/Solution.cs
// LeetCode 2526 - Find Consecutive Integers from a Data Stream
// https://leetcode.com/problems/find-consecutive-integers-from-a-data-stream/

public class DataStream {
    private int value, k, streak;

    public DataStream(int value, int k) {
        this.value = value;
        this.k = k;
        streak = 0;
    }

    public bool Consec(int num) {
        if (num == value) streak++;
        else streak = 0;
        return streak >= k;
    }
}

File: 2526_find_consecutive_integers_from_a_data_stream/solution.php
<?php
// LeetCode 2526 - Find Consecutive Integers from a Data Stream
// https://leetcode.com/problems/find-consecutive-integers-from-a-data-stream/

class DataStream {
    private $value;
    private $k;
    private $streak;

    function __construct($value, $k) {
        $this->value = $value;
        $this->k = $k;
        $this->streak = 0;
    }

    function consec($num) {
        if ($num === $this->value) $this->streak++;
        else $this->streak = 0;
        return $this->streak >= $this->k;
    }
}
